### DISCUSSION TITLE
How does bullmq ensure jobs aren’t deleted by Redis evictions?

### DISCUSSION THREAD

**Discussion by danthegoodman1** [taskforcesh/bullmq#420] [Q&A]
Since Redis can evict kv pairs for various reasons, would this potentially cause jobs to be removed before execution? For example a repeating job or delayed job getting evicted before executing.

**Comment by manast** [comment]
You should not run Redis with any other eviction policy than "noeviction", which is the default one.

**Reply by stephenhuh** [reply]
can you clarify on this piece? doesn't that mean we'd just run out of memory with enough jobs?

for usage with bullmq does this mean we are expected to delete our own completed jobs?

**Reply by manast** [reply]
You need to dimension your Redis instance according to your expectations, activate a Redis memory alert monitor for around 75% and have proper removeOnComplete/Faile policies to keep the memory usage below limits.
